openwisp_notifications/base/admin.py

```diff
@@ -15,14 +15,28 @@ def get_readonly_fields(self, request, obj=None):
         else:
             return self.readonly_fields
 
+    def get_fields(self, request, obj=None):
+        fields = super().get_fields(request)
+        if not request.user.is_superuser and 'deleted' in fields:
+            fields.remove('deleted')
+        return fields
+
     def has_add_permission(self, request, obj=None):
         return request.user.is_superuser
 
     def has_delete_permission(self, request, obj=None):
-        return request.user.is_superuser
+        # A notification is considered deleted when the delete
+        # boolean field is set. A user can never delete
+        # NotificationSettting from the database.
+        return False
```

Comment:
are you sure this won't prevent us from deleting users? Did you try to delete users? I remember in the past we had problems when setting this to False (we couldn't delete the related objects from the admin).